src/frontend/apps/standalone_site/src/features/Playlist/features/UpdatePlaylist/components/AddUserAccessForm.tsx

```diff
@@ -0,0 +1,165 @@
+    description: 'Select role for user to add in playlist.',
+    id: 'features.Playlist.features.UpdatePlaylist.components.AddUserAccessForm.roleSelectLabel',
+  },
+  addMember: {
+    defaultMessage: 'Add this member',
+    description: 'Add member to playlist button title.',
+    id: 'features.Playlist.features.UpdatePlaylist.components.AddUserAccessForm.addMember',
+  },
+  searchInfos: {
+    defaultMessage:
+      'Search within user and select the one you want to add to this playlist.',
+    description: 'Modale description to search user to add into playlist.',
+    id: 'features.Playlist.features.UpdatePlaylist.components.AddUserAccessForm.searchInfos',
+  },
+  placeHolder: {
+    defaultMessage: 'Search by username or email address...',
+    description: 'Search user input placeholder.',
+    id: 'features.Playlist.features.UpdatePlaylist.components.AddUserAccessForm.placeHolder',
+  },
+});
+
+interface AddUserAccessFormProps {
+  playlistId: string;
+  onUserAdded: () => void;
+  excludedUsers?: string[];
+}
+
+export const AddUserAccessForm = ({
+  playlistId,
+  onUserAdded,
+  excludedUsers,
+}: AddUserAccessFormProps) => {
+  const intl = useIntl();
+
+  const options = useMemo(() => userRoleOptions(intl), [intl]);
+  const [typedStr, setTypedStr] = useState('');
+  const [searchedUser, setSearchedUser] = useState('');
+  const [selectedUser, setSelectedUser] = useState<Nullable<UserLite>>(null);
+  const [roleValue, setRoleValue] = useState(
+    options.find((option) => option.key === PlaylistRole.STUDENT),
+  );
+  const { mutate: createPlaylistAccess } = useCreatePlaylistAccess({
+    onSuccess: () => {
+      onUserAdded();
+    },
+  });
+
+  const debouncedSetSearchedStr = useRef(
+    debounce((str: string) => setSearchedUser(str), 1000),
+  ).current;
+
+  // Prevent debounce leak
+  useEffect(() => {
+    return () => {
+      debouncedSetSearchedStr.cancel();
+    };
+  }, [debouncedSetSearchedStr]);
+
+  return (
+    <Box>
+      <Heading alignSelf="center">{intl.formatMessage(messages.title)}</Heading>
+
+      {selectedUser && (
+        <Box gap="small" align="start">
+          <Box flex={false} fill={false}>
+            <Button
+              plain
+              icon={<PreviousIcon />}
+              label={intl.formatMessage(messages.back)}
+              onClick={() => {
+                setSelectedUser(null);
+              }}
+            />
+          </Box>
+          <SearchUserListRow user={selectedUser}>
+            <Select
+              a11yTitle={intl.formatMessage(messages.roleSelectLabel)}
+              labelKey="label"
+              value={roleValue}
+              options={options}
+              onChange={({
+                option,
+              }: {
+                option: { label: string; key: PlaylistRole };
+              }) => {
+                setRoleValue(option);
+              }}
+            />
+          </SearchUserListRow>
+          <Button
```

Comment:
It exists the component `ModalButton`, same to keep consistency between design.

https://github.com/openfun/marsha/blob/feature/vassili/create-playlist-accesses/src/frontend/apps/standalone_site/src/features/Contents/features/ClassRoom/components/Create/ClassRoomCreateForm.tsx#L218